File: python/plan_llegar_trabajo.py
# importamos la funciones para verificar si una entrada es un numero
from verificar_numero import esNumeroPositivo, solicitarNumeros

print('===Opciones para llegar a tiempo al trabajo ===')

hora= solicitarNumeros("Ingrese la hora que tiene planeada para salir de su casa a el trabajo: ")
tipoTransporte=""
repeticion = 0
# validamos que el usuario ingrese alguna de las dos opciones existentes
while tipoTransporte not in ['b','c'] :
    if repeticion <5:
        print("""Escoja tipo de transporte:
        b = bus
        c =carro
        """)
        tipoTransporte= (input('Elija su opcion: ')).lower()
        repeticion+=1
    else:
        raise TypeError("Solo se acepta opcion b o c, has superado el numero de intentos permitidos")
# formula para calcular tiempo en llegar al trabajo
time_en_llegar= 9 - hora
# tiempo en llegar al trabajo en bus
time_llegar_bus=2
# tiempo en llegar al trabajo en carro
time_llegar_carro=1
time_extra_bus=time_en_llegar-time_llegar_bus
time_extra_carro=time_en_llegar-time_llegar_carro

if time_en_llegar < 2 and  tipoTransporte =='b':
    print('Para llegar a tiempo en bus debes salir con minimo 2 horas de anticipo')
elif time_en_llegar < 1 and tipoTransporte == 'c':
    print('Para llegar a tiempo en carro debes llegar con minimo 1 hora de anticipo')
elif time_en_llegar >2 and tipoTransporte == 'b':
    print('Es una buena opcion llegaras con ',time_extra_bus,' horas de anticipo')
elif time_en_llegar >1 and tipoTransporte == 'c':
    print('Es una buena opcion llegaras con ',time_extra_carro,' horas de anticipo')

File: problemas_algoritmos/javascript/plan_llegar_trabajo.js

function crearPlan(){
    const inputHoraEntrada = document.querySelector("#horaEntrada");
const horaEntrada = esNumeroPositivo( inputHoraEntrada.value);
const inputHoraSalida = document.querySelector("#horaSalida");
const horaSalida = esNumeroPositivo( inputHoraSalida.value);
const sb = document.querySelector("#transporte");
// const btn = document.querySelector("#btn")

const selectedValues = [].filter
                .call(sb.options, option => option.selected)
                .map(option => option.text);
console.log(selectedValues)
let time_llegar = horaEntrada - horaSalida;
const time_llegar_bus = 3;
const time_llegar_carro = 2;
time_extra_bus=time_llegar-time_llegar_bus
time_extra_carro=time_llegar-time_llegar_carro
if(time_llegar < 3 && selectedValues == 'Bus'){
    alert('Para llegar a tiempo en bus debes salir con minimo 3 horas de anticipo')
}else if(time_llegar < 2 && selectedValues == 'Carro'){
    alert('Para llegar a tiempo en carro debes salir con minimo 2 hora de anticipo')
}else if(time_llegar > 3 && selectedValues == 'Bus'){
    alert('Es una buena opcion llegaras con '+ time_extra_bus +' horas de anticipo')
}else if(time_llegar > 2 && selectedValues == 'Carro'){
    alert('Es una buena opcion llegaras con '+time_extra_carro+' horas de anticipo')
}else if(time_llegar = 3 && selectedValues == 'Bus'){
    alert('Llegaras justo a tiempo')
}else if(time_llegar = 2 && selectedValues == 'Carro'){
    alert('Llegaras jussto a tiempo')
}
}
function clean(){
    const inputhoraEntrada = document.querySelector("#horaEntrada");
    inputhoraEntrada.value = "";
    const inputhoraSalida = document.querySelector("#horaSalida");
    inputhoraSalida.value = "";
    const inputTipoTransporte = document.querySelector("#transporte");
    inputTipoTransporte.value = "";
}
